tracer/src/Datadog.Trace/DuckTyping/DuckTypeExtensions.cs

```diff
@@ -168,12 +139,8 @@ public static bool DuckIs<T>(this object? instance)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool DuckIs(this object? instance, Type? targetType)
         {
-            if (instance is null)
-            {
-                DuckTypeTargetObjectInstanceIsNull.Throw();
-            }
-
-            if (targetType != null)
+            if (instance is not null &&
+                targetType is not null)
```

Comment:
It seems like a programming error for `targetType` to be null here 🤔

Reply:
I did some changes in https://github.com/DataDog/dd-trace-dotnet/pull/4582/commits/fbb089537a2198a51d6a7faf802de611993dc3c6